src/prefect/engine.py

```diff
@@ -370,12 +372,31 @@ async def create_then_begin_flow_run(
         )
 
     if return_type == "state":
-        return state
+        res = state
     elif return_type == "result":
-        return await state.result(fetch=True)
+        res = await state.result(fetch=True)
     else:
         raise ValueError(f"Invalid return type for flow engine {return_type!r}.")
 
+    if PREFECT_VIZ_MODE:
+        task_runs = await client.graph(flow_run.id)
+
+        id_to_name = {task_run["id"]: task_run["name"] for task_run in task_runs}
+        edges = []
+        for task_run in task_runs:
+            for dependency in task_run["upstream_dependencies"]:
+                if dependency["input_type"] == "task_run":
+                    name = id_to_name[dependency["id"]]
+                    edges.append((name, task_run["name"]))
+
+        g = graphviz.Digraph("G", filename=f"{flow.name}")
+        for edge in edges:
+            g.edge(*edge)
+
+        g.view()
```

Comment:
I think .render() might provide more flexibility? We can specify a filename, format, etc.

e.g. 
```python
g.render('flow_graph.png', view=True)
```